src/main/scala/io/phdata/retirementage/loadgen/LoadGenerator.scala

```diff
@@ -27,13 +43,71 @@ object LoadGenerator {
     * @param payloadBytes Payload size
     * @return The test dataframe
     */
-  def generateTable(numRecords: Int, payloadBytes: Int): DataFrame = ???
+  def generateTable(spark: SparkSession, numRecords: Int, payloadBytes: Int): DataFrame = {
+
+    /**
+      * Create the following schema:
+      *   -- id: String (nullable = false)
+      *   -- payload: String (nullable = false)
+      *   -- date: String (nullabe = false)
+      */
+    val schema = StructType(StructField("id", StringType, false) :: Nil)
+      .add(StructField("payload", StringType, false))
+      .add(StructField("date", StringType, false))
+    var dataBuffer = ListBuffer[List[String]]()
+    // Create a string with specified bytes
+    var i          = 0
+    var byteString = ""
+    for (i <- 1 to payloadBytes) {
+      byteString = byteString + "a"
+    }
+    // Create numRecords rows of data with a random UUID and attached a payload of specified bytes and a date
+    var j = 0
+    for (j <- 1 to numRecords) {
+      dataBuffer += List(UUID.randomUUID().toString.substring(0, 9),
+                         byteString,
+                         tempDateGenerator(j))
+    }
+    val dataList = dataBuffer.toList
+    // Creating rows and RDD for the DataFrame
+    val rows = dataList.map(x => Row(x: _*))
+    val rdd  = spark.sparkContext.makeRDD(rows)
+
+    // Returning DataFrame
+    spark.createDataFrame(rdd, schema)
+  }
+
+  // Create date data with 2016-12-25, 2017-12-25, 2018-12-25
+  def tempDateGenerator(num: Int): String = {
+    val r = new scala.util.Random()
+    val c = 0 + r.nextInt(3)
+    c match {
+      case 0 => "2016-12-25"
+      case 1 => "2017-12-25"
+      case 2 => "2018-12-25"
+      case _ => "2222-22-22"
+    }
+  }
 
   /**
     * Generates a test dataframe with keys from the parent used as foreign keys
     * @param numRecords Number of records in the dataframe
     * @param payloadBytes Payload size
     * @return The test dataframe
     */
-  def generateTableFromParent(numRecords: Int, payloadBytes: Int, parent: DataFrame): DataFrame = ???
+  def generateTableFromParent(spark: SparkSession,
+                              numRecords: Int,
+                              payloadBytes: Int,
+                              parent: DataFrame): DataFrame = {
+    var i = 0
+    var byteString = ""
+    for (i <- 1 to payloadBytes) {
+      byteString = byteString + "a"
+    }
+
+    // Can the numRecords for the child be larger than the parent?
```

Comment:
Yes we should be able to build a child that's bigger than a parent. We can use the same unioning method as above.